core/lib/eth_signer/src/g_kms_signer.rs

```diff
@@ -0,0 +1,174 @@
+use std::result::Result;
+
+use ethers_signers::Signer as EthSigner;
+use google_cloud_auth::credentials::CredentialsFile;
+use google_cloud_gax::retry::RetrySetting;
+use google_cloud_kms::{
+    client::{google_cloud_auth, Client, ClientConfig},
+    signer::ethereum::Signer,
+};
+use hex;
+use tracing::{self};
+use zksync_types::{
+    web3::{keccak256, Signature},
+    Address, EIP712TypedStructure, Eip712Domain, PackedEthSignature, H256, U256,
+};
+
+use crate::{
+    raw_ethereum_tx::{Transaction, TransactionParameters},
+    EthereumSigner, SignerError,
+};
+
+#[derive(Clone)]
+pub struct GKMSSigner {
+    signer: Signer,
+}
+
+impl GKMSSigner {
+    pub async fn new(key_name: String, _chain_id: u64) -> Result<Self, SignerError> {
+        tracing::info!(
+            "KMS signer credentail path: {:?}",
+            std::env::var("GOOGLE_APPLICATION_CREDENTIALS")
+        );
+
+        let credentials_path = std::env::var("GOOGLE_APPLICATION_CREDENTIALS").map_err(|_| {
+            SignerError::SigningFailed(
+                "Environment variable GOOGLE_APPLICATION_CREDENTIALS not found".to_string(),
+            )
+        })?;
+
+        let cf = CredentialsFile::new_from_file(credentials_path)
```

Comment:
The google auth accept several types of Credentials. I've described in the KMS deployment note. The credential file will be stored in `/home/.config/glocund/xxx_credentails.json` by default. [attach a service account to a resource](https://cloud.google.com/iam/docs/Attach-service-accounts#attaching-to-resources)  might be the best practice to avoid store secrets in the machine. Maybe Devops can help me to setup/configure it because I don't think I have permission to touch the google computer engine resources.

However, if we run it in a confidential VM, will it be a concerns?

Reply:
Thanks @JayT106 , I need to understand more about the implication of attaching service account to a resource before commenting. Does attaching the service account means the compute engine can call to the KMS sign function without any authentication?

In this case, this may be even worse than a credential file because anyone who gain access to the machine can gain access immediately (if I am not mistaken). Having a file, would still lengthen the time of locating the file and gaining access perhaps.

Using a TEE should generally be better. But I will need to read a bit more to comment on a secure setup.

Reply:
Resolving this for now. Will reach out separately after I study the Confidential VM.